dracut/30afterburn/afterburn-network-kargs.service

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Afterburn Initrd Setup Network Kernel Arguments
+
+# This service may produce additional kargs fragments,
+# which are then consumed by dracut-cmdline(8).
+DefaultDependencies=no
+Before=dracut-cmdline.service
+Before=ignition-fetch.service
+After=systemd-journald.socket
+
+OnFailure=emergency.target
+OnFailureJobMode=isolate
+
+[Service]
+Environment=AFTERBURN_OPT_PROVIDER=--cmdline
+# AFTERBURN_NETWORK_KARGS_DEFAULT must be set externally by distributions.
+ExecStart=/usr/bin/afterburn exp rd-network-kargs ${AFTERBURN_OPT_PROVIDER} --default-value $AFTERBURN_NETWORK_KARGS_DEFAULT
```

Comment:
Shouldn't this be using `${}` too since we want it passed as a single argument?

Reply:
i.e. to handle the case where the default is `ip=foo ip=bar nameserver=host`?

Reply:
Almost, but no. It's both tricky and subtle, so I'll note here all the details. The key part is in https://github.com/systemd/systemd/pull/15450.

`${FOO}` always expands to something (e.g. an empty string), even if the argument is missing. But @cgwalters wants an explicit failure if the default is missing/typoed on the distro side. And an empty string is a totally valid default. So we resort to `$FOO` which expands to zero or more values. Getting one value out of a whitespaced string is achieved by enclosing the value in quotes at the injection point (i.e. the distro dropin).

Reply:
Looks like a good comment to put in the unit file so someone doesn't try to change it in the future.